resource-managers/nomad/src/main/scala/org/apache/spark/scheduler/cluster/nomad/ExecutorTask.scala

```diff
@@ -72,10 +71,9 @@ private[spark] object ExecutorTask
       .map(Utils.splitCommandString).getOrElse(Seq.empty)
 
     task.addEnv("SPARK_EXECUTOR_OPTS", (extraJavaOpts ++ executorConf).mkString(" "))
-
     task.addEnv("SPARK_EXECUTOR_MEMORY", jvmMemory(conf, task))
 
-    val sparkLocalDirs = conf.getOption("spark.local.dir").getOrElse("/spark-local-dir")
+    val sparkLocalDirs = conf.getOption("spark.local.dir").getOrElse("/tmp")
```

Comment:
this code previously set `SPARK_LOCAL_DIRS` to the nomad allocation directory, which was shared between the executor task and shuffle service task in the same task group.  Now we are making it configurable and defaulting to `/tmp` (which is default behavior in core Spark) since we can no longer use the shared task group directory.  When using Docker the user will need to volume mount this directory onto their Nomad hosts and ensure the external shuffle service has access to that directory. The nomad-specific docs have been updated to reflect this.